content/WebAuthn/WebAuthn_Developer_Guide/user-handle.adoc

```diff
@@ -0,0 +1,61 @@
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+
+
+== User Handle ==
+
+
+:toc:
+:toc-placement!:
+toc::[]
+
+This description is derived in large part from the W3C's recommendation, https://www.w3.org/TR/webauthn/[Web Authentication: An API for accessing Public Key Credentials Level 1].
+
+The user handle represents the mapping of a public key credential to a user account with the Relying Party (RP). As such, it is the RP that sets its value, ``user.id``.
+
+*Characteristics of ``user.id``*
+
+* Must not contain information that could identify the user.
+* Opaque byte sequence, maximum length = 64
+* Must be pseudo random
+* Must not start at 1 or 0
+
+An authenticator maps the user handle to the public key credential source. Authentication and authorization decisions are made on the basis of the ``user.id`` (not the user handle, which can be understood as a nickname in the sense of https://tools.ietf.org/html/rfc8266#section-6.1[`RFC 8266]).  The ``user.id`` and other user account attributes reside in the ``PublicKeyCredentialUserEntity`` dictionary, which is used when creating a new credential, as shown here:
+
+[source]
+----
+{
+   dictionary PublicKeyCredentialUserEntity : PublicKeyCredentialEntity {
+       required BufferSource   id;
+       required DOMString      displayName;
+   };
+}
+----
+
+The user handle is required for usernameless passwordless authentication with resident credentials, i.e., credentials that live on the security key. The example below illustrates how registration of the key creates the ``user.id`` underlying the user handle that is verified during authorization. In this example, at no point after registration is either a username or a password required.
+
+[NOTE]
+====
+Universal second factor authentication (U2F) does not support the user handle.
+====
+
+== Example
+When user John registers his security key with a Microsoft online service (the RP), the service prompts John to allow the service to create and register a credential on the key. That service identifies the pairing (the public key credential on John's key + John's account)  by means of the user handle, by setting the value for the ``user.id`` to the following, for example:
+
+[source]
+----
+{
+   user.id = 69 b3 72 54 75 20 ee 1f 91 7e 62 06 86 87 18 94
+   02 94 1f f0 31 75 d7 d9 91 7c d5 f6 2c c3 cd 39
+   b2 f9 ee 15 2f 13 8a f7 fa 29 bc fe ab 35 c4 aa
+   89 40 b5 54 c1 83 4a 85 fa 8a dd d5 f6 6c a1 0d
+}
+----
+
+When John logs in using the security key, the authenticator verifies that the public key credential on that particular key is associated with a specific account, and that the source of the public key credential on John's key was indeed Microsoft.
```

Comment:
>the authenticator verifies that the public key credential on that particular key is associated with a specific account

This isn't quite true. If `allowCredentials` is set (presumably to the keys for only one account), then the authenticator uses one of the credentials listed therein; but if not (usernameless login), the authenticator can't know what account it would check for, so it can use any credential bound to the RP.

>the authenticator verifies [...] that the source of the public key credential on John's key was indeed Microsoft

I think this can be expressed more clearly as something like "the key was created for Microsoft", to not risk suggesting that the private key is creted by Microsoft.

```suggestion
When John logs in using the security key, the authenticator verifies that the public key credential on John's key was indeed created for Microsoft, and returns the user handle along with the authentication signature. Microsoft can then look up the user handle in their user database and conclude that the person logging in is John.
```